ci/kind/kind-setup.sh

```diff
@@ -300,6 +300,18 @@ function configure_vlan_subnets {
       docker_run_with_host_net iptables -t filter -A FORWARD -i ${vlan_interfaces[j]} -o ${vlan_interfaces[i]} -j ACCEPT
     done
   done
+
+  # Adding ipset rules to prevent SNAT based on flexibleIPAM e2e tests
+  if [[ $FLEXIBLE_IPAM == true ]]; then
+    docker_run_with_host_net ipset create excluded_subnets hash:net
+    docker_run_with_host_net ipset add excluded_subnets 192.168.241.0/24
+    docker_run_with_host_net ipset add excluded_subnets 192.168.242.0/24
+    docker_run_with_host_net ipset add excluded_subnets 192.168.240.0/24
+    docker_run_with_host_net ipset list excluded_subnets
+    
+    docker_run_with_host_net iptables -t nat -I POSTROUTING 1 ! -o  $bridge_interface -s 192.168.240.0/24 -m set --match-set excluded_subnets dst -j RETURN
+    docker_run_with_host_net iptables -t nat -A POSTROUTING  ! -o $bridge_interface -s 10.244.0.0/16 -m set ! --match-set excluded_subnets dst -j MASQUERADE
```

Comment:
when using FlexibleIPAM, noSNAT is set to true and Antrea will not perform SNAT for Pod traffic
this rule seems to counteract that behavior, by reestablishing SNAT for regular Pods (which use the default Pod CIDR). Can you explain why it is needed?

Reply:
what I observe that when we create kind network then docker will add corresponding default NAT rule in the iptables:
`Chain POSTROUTING (policy ACCEPT 887 packets, 63828 bytes)

 pkts bytes target     prot opt in     out     source               destination         

    0     0 MASQUERADE  0    --  *      !br-053bd7f613f8  192.168.240.0/24     0.0.0.0/0           

    0     0 MASQUERADE  0    --  *      !docker0  172.17.0.0/16        0.0.0.0/0           

    0     0 MASQUERADE  6    --  *      *       192.168.240.2        192.168.240.2        tcp dpt:6443`
    
and because of that it's performing SNAT  for the case:https://github.com/antrea-io/antrea/issues/6237, and failing  tests, to prevent that i added two NAT rules.

Reply:
I don't think that's answering my question.
The purpose of the first rule is to prevent SNAT between VLANs and I understand that. The second rule on the other hand is forcing SNAT for "regular" (NodeIPAM) Pods, so I am asking you to clarify why it's needed with a comment. I'm assuming something like this is correct:

> With FlexibleIPAM, Antrea SNAT is disabled (noSNAT: true) so Pods don't have access to the external network by default (including regular / NodeIPAM Pods). With our configuration, Docker will already perform SNAT for FlexibleIPAM Pods. However, we need a custom SNAT rule for regular Pods.

I am not sure that the Pods really need access to the external network, but maybe some tests break without this?

Reply:
for some of the tests cases it's need external connectivity:
```
I0203 08:40:22.644633   26656 k8s_util.go:230] z-51un7txt/a -> facebook.com: expected Con but got Drp: err - <nil> /// stdout -  /// stderr - 1: TIMEOUT

2: TIMEOUT

3: TIMEOUT

FINISHED
```

Reply:
Yes, that's what I figured. Please keep the rule and add the suggest comment ^